spec/controllers/settings/dex_connector_oidcs_controller_spec.rb

```diff
@@ -0,0 +1,371 @@
+      get :edit, id: DexConnectorOidc.last.id + 1
+      expect(response).to have_http_status(:not_found)
+    end
+
+  end
+
+  # rubocop:disable RSpec/ExampleLength
+  describe "GET #create" do
+    good_provider = "http://your.fqdn.here:5556/dex"
+
+    it "fails validation with a bad provider" do
+      get :create, validate: true, dex_connector_oidc: {
+        name:          "good oidc",
+        provider_url:  "dead",
+        callback_url:  "http://well.formed.but.invalid/",
+        basic_auth:    true,
+        client_id:     "client",
+        client_secret: "secret"
+      }
+      expect(response).to render_template("new")
+      expect(assigns(:is_data_valid)).to be(false)
+    end
+
+    it "passes validation with a good provider" do
+      VCR.use_cassette("oidc/validate_connector", allow_playback_repeats: true, record: :none) do
+        get :create, validate: true, dex_connector_oidc: {
+          name:          "good oidc",
+          provider_url:  good_provider,
+          callback_url:  "http://well.formed.but.invalid/",
+          basic_auth:    true,
+          client_id:     "client",
+          client_secret: "secret"
+        }
+      end
+      expect(response).to render_template("new")
+      expect(assigns(:is_data_valid)).to be(true)
+    end
+
+    it "fails creating with a bad provider" do
+      get :create, validate: false, dex_connector_oidc: {
+        name:          "good oidc",
+        provider_url:  "dead",
+        callback_url:  "http://well.formed.but.invalid/",
+        basic_auth:    true,
+        client_id:     "client",
+        client_secret: "secret"
+      }
+      expect(response).to render_template("new")
+      expect(response).to have_http_status(:unprocessable_entity)
+    end
+
+    it "passes creating with a good provider" do
+      VCR.use_cassette("oidc/validate_connector", allow_playback_repeats: true, record: :none) do
+        get :create, validate: false, dex_connector_oidc: {
+          name:          "good oidc",
+          provider_url:  good_provider,
+          callback_url:  "http://well.formed.but.invalid/",
+          basic_auth:    true,
+          client_id:     "client",
+          client_secret: "secret"
+        }
+      end
+      expect(response).not_to have_http_status(:unprocessable_entity)
+      # expect(subject).to redirect_to(settings_dex_connector_oidcs_path)
+      expect(response).to redirect_to(settings_dex_connector_oidcs_path)
+    end
+
+    context "with rendered views" do
+      render_views
+
+      it "shows an error message for empty name" do
+        VCR.use_cassette("oidc/validate_connector", allow_playback_repeats: true, record: :none) do
+          get :create, validate: false, dex_connector_oidc: {
+            name:          "",
+            provider_url:  good_provider,
+            callback_url:  "http://well.formed.but.invalid/", # can't be blank in form
+            basic_auth:    true, # also can't be blank - only true/false
```

Comment:
There doesn't appear to be any test with basic_auth set to false. There should be a test to ensure if it is false it outputs as false.

Reply:
Pillar test added in 78ace6e.  More useful tests "coming soon".

Reply:
Upon further discussion, the pillar test is sufficient, as that's the only place it's consumed.  Any further testing would be in Salt and Dex.